src/http/httpd.cc

```diff
@@ -389,14 +389,27 @@ void http_server::set_content_streaming(bool b) {
     _content_streaming = b;
 }
 
-future<> http_server::listen(socket_address addr, listen_options lo) {
-    if (_credentials) {
-        _listeners.push_back(seastar::tls::listen(_credentials, addr, lo));
+future<> http_server::listen(socket_address addr, listen_options lo,
+            shared_ptr<seastar::tls::server_credentials> listener_credentials) {
+    if (listener_credentials) {
+        _listeners.push_back(seastar::tls::listen(listener_credentials, addr, lo));
     } else {
         _listeners.push_back(seastar::listen(addr, lo));
     }
     return do_accepts(_listeners.size() - 1);
```

Comment:
Is that correct? If I call `lister(addr, lo, creds)` it will add tls-enabled listener and will call `do_accepts(nr)`. The latter, in turn, would call `do_accepts(nr, _credentials != nullptr)` which is wrong -- `_credentials` shouldn't be used in this particular case

Reply:
you are right, sorry for this, it should actually decide based on the present of current listener credentials, fixed